controllers/components/kueue/kueue_controller.go

```diff
@@ -67,6 +69,20 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 		Owns(&admissionregistrationv1.MutatingWebhookConfiguration{}).
 		Owns(&admissionregistrationv1.ValidatingWebhookConfiguration{}).
 		Owns(&appsv1.Deployment{}, reconciler.WithPredicates(resources.NewDeploymentPredicate())).
+		Watches(
+			&extv1.CustomResourceDefinition{},
+			reconciler.WithEventHandler(
+				handlers.ToNamed(componentApi.KueueInstanceName)),
+			reconciler.WithPredicates(predicate.Or(
+				component.ForLabel(labels.ODH.Component(LegacyComponentName), labels.True),
+				// For the case when user manually delete old MultiKueueConfig/MultiKueueCluster CRDs v1alpha1
+				predicate.Funcs{
+					DeleteFunc: func(e event.DeleteEvent) bool {
+						return e.Object.GetName() == gvk.MultiKueueConfigV1Alpha1.Kind || e.Object.GetName() == gvk.MultikueueClusterV1Alpha1.Kind
```

Comment:
this should probably already covered by the `component.ForLabel` predicate (https://github.com/opendatahub-io/opendatahub-operator/blob/main/pkg/controller/predicates/component/component.go#L15-L17), unless the crd were not deployed by the operator.

Reply:
right, if the CRD has `app.opendatahub.io/kueue: 'true'`

i dont know if we should cover it or not, for the case, kueue team can run "make deploy" to get CRD into cluster but these wont have labels/annotations.
maybe we can remove `DeleteFuncs` here